hazelcast/src/main/java/com/hazelcast/cache/impl/AbstractCacheService.java

```diff
@@ -288,6 +289,7 @@ public void deleteCache(String name, String callerUuid, boolean destroy) {
         CacheConfig config = deleteCacheConfig(name);
         if (destroy) {
             destroySegments(name);
+            sendInvalidationEvent(name, null, SOURCE_NOT_AVAILABLE);
```

Comment:
Yes, this is necessary, otherwise `Cache.destroy` and `CacheManager.destroyCache`
will not deliver an invalidation event (which used to be the case until 3.7).

The method usage in `reset` invokes `deleteCache` with argument `destroy = false`,
so there will be no duplicate invalidation event.